myapp/app/controllers/admin/labels_controller.rb

```diff
@@ -0,0 +1,51 @@
+class Admin::LabelsController < ApplicationController
+  before_action :check_admin_auth
+
+  def index
+    @labels = Label.all
+  end
+
+  def new
+    @label = Label.new
+  end
+
+  def edit
+    @label = Label.find(params[:id])
+  end
+
+  def create
+    @label = Label.new(label_params)
+    if @label.save
+      flash[:success] = t('label.flashes.success.created')
+      redirect_to admin_labels_path
+    else
+      render :new
+    end
+  end
+
+  def update
+    @label = Label.find(params[:id])
+    if @label.update(label_params)
+      flash[:notice] = t('label.flashes.success.updated')
+      redirect_to admin_labels_path
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @label = Label.find(params[:id])
+    puts(@label.tasks)
+    @label.tasks.destroy_all
```

Comment:
Is it OK to remove all the tasks related to the label?
I just thought that it was a little bit weird

Reply:
When 2 labels were set to the task, and 1 was deleted, that task will be also deleted...

Reply:
umm... isn't it about the pivot table entry removing?

suppose we have 2 labels attached to a single task. and the entry's are in `tasks_labels` joint table. as `{task_id => 1, label_id = 1, task_id => 1, label_id = 2}` so removing `{label_id = 2}` will only remove from `tasks_labels` not from `tasks` table

Reply:
Oh, I see, thanks for explaining!